src/components/LocaleLink.js

```diff
@@ -21,8 +21,8 @@ const LocaleLink = ({ children, to, locale, ...other }) =>
       }
 
       localeTo = localeTo.replace(
-        new RegExp(`^/${supportedLocales.join('|')}(/.*)$`),
-        `${locale === defaultLocale ? '' : '/' + locale}$1`
+        new RegExp(`^/${supportedLocales.join('/|/')}/(.*)$`),
+        locale === defaultLocale ? '/' : `/${locale}/`
```

Comment:
Oh no but now it’ll eat the end of the URL right?

Reply:
I think it's fine? Not sure what you mean?
This `/de/developers/tbtc-technical-system-overview/node/de/test` becomes `/developers/tbtc-technical-system-overview/node/de/test`?

Reply:
Hmm. I’m puzzled because we still match through the end of the URL but we no longer have a $1 backreference to include it in the replacement. I’ll poke it today to fix my understanding.

Reply:
Oh goodness. Man this regex was *really broken* <_<

Here's the scenario where this version breaks:

```
> const supportedLocales = ["en","de","es","ru"]
< undefined
> const defaultLocale = "en"
< undefined
> ((locale) => "/ru/booyan/slapadap/1234-123-gap-map-de".replace(new RegExp(`/${supportedLocales.join('/|/')}/(.*)$`), locale === defaultLocale ? '/' : `/${locale}/`))("en")
< "/"
```

If we look at the final regex, it's this:

```
/\/en\/|\/de\/|\/es\/|\/ru\/(.*)$/
```

And the old one:

```
/^\/en|de|es|ru(\/.*)$/
```

None of these actually does what we want, because the alternation operator (`|`) doesn't care what you *want* the alternates to be, it cares what the regex says they are. If we break the above down, here's how they map:

```
/\/en\/|\/de\/|\/es\/|\/ru\/(.*)$/
  -> /en/ OR /de/ OR /es/ OR /ru/<the rest of the string>
/^\/en|de|es|ru(\/.*)$/
  -> /en OR de OR es OR ru/<the rest of the string>
```

In the old case, we only matched the leading `/` for `en`, and we only pulled out the rest of the string for `ru` (or whatever the last entry in the `supportedLocales` array was). The replacement included the `$1` backreference, so it correctly handled the last locale, at the cost of completely dropping the ball on all non-English locales for the leading `/` matcher. In the new case, we properly match the bracketing `/`es on all locales, but in the last locale we also match the rest of the string. Because the `$1` backreference is no longer there, however, that last locale will have the rest of its URL eaten.

The site's had broken links for a bit longer than ideal, so I'm going to drop the `.*` matcher and merge. I don't think that matcher is necessary here---JavaScript regexes used in `replace` only match once by default (you have to pass the `g` flag to match multiple times), which I think will have the effect the `(.*)` was going for here.

We really need to give this a bit more thorough testing at some point to make sure nothing else is broken.

You know what they say about regular expressions :wink:

Reply:
Actually I'm still not entirely sure what this is trying to do. I think its only purpose is replacing the default locale with a blank? In which case this can just be `.replace("/" + defaultLocale + "/", "")` or so, I think.